test/sdc-api.spec.js

```diff
@@ -1,21 +1,92 @@
+/* eslint-disable no-unused-vars */
 /* eslint-disable no-undef */
 import 'regenerator-runtime/runtime';
 
+jest.useFakeTimers();
+
 const supertest = require('supertest');
 const app = require('../server/server');
 
 const request = supertest(app);
 
 describe('API calls', () => {
-  test('/sidebar/all route works for existing courseId', async (done) => {
+  test('GET /sidebar/all route works for existing courseId', async (done) => {
+    expect.assertions(3);
     request.get('/sidebar/all?courseId=1')
-      .expect('Content-Type', 'application/json; charset=utf-8')
-      .expect(200)
       .then((response) => {
-        expect(response.body.price.basePrice).toEqual(99.99);
+        expect(response.status).toBe(200);
+        expect(response.body.price.basePrice).toEqual(89.99);
         expect(response.body.sidebar.fullLifetimeAccess).toEqual('Full lifetime access');
         done();
       })
       .catch((err) => done(err));
   });
+
+  test('POST /sidebar/all route works for new courseId', async (done) => {
+    const testData = {
+      courseId: 101,
+      price: {
+        basePrice: 99.99,
+        discountPercentage: 84,
+        discountedPrice: 15.99,
+        saleEndDate: '2021-06-29T20:14:35.238Z',
+        saleOngoing: false,
+      },
+      sidebar: {
+        fullLifetimeAccess: 'Full lifetime access',
+        accessTypes: 'Access on mobile and TV',
+        assignments: true,
+        certificateOfCompletion: true,
+        downloadableResources: 20,
+      },
+      previewVideo: {
+        previewVideoImgUrl: 'http://localhost:3004/assets/previewVideoImg6.jpg',
+        previewVideoUrl: 'http://localhost:3004/assets/previewVideo6.mp4',
+      },
+    };
```

Comment:
Since you typed a lot here, maybe you can replace the object to be inserted by calling data generation function if that is available.

Reply:
yeah, I was planning on doing that for when I do my data generation script. just trying to work with what legacy code gave me and get meaningful test through. but this definitely makes sense and I'll implement it for my datagenerations.